guides/yaml.rst

```diff
@@ -0,0 +1,298 @@
+      friendly_name: $friendly_name
+
+    sensor:
+      - platform: dht
+        model: DHT22
+        pin: D2
+        temperature:
+          name: "${friendly_name} Temperature"
+          id: ${device_name}_temperature
+        humidity:
+          name: "${friendly_name} Humidity"
+        update_interval: ${update_interval}
+
+**Key Features:**
+
+- Values are referenced using ``$variable_name`` or ``${variable_name}`` syntax
+- The ``${variable_name}`` syntax is required when embedding within other text
+- Substitutions are processed before any other part of the configuration
+- Substitutions values must be strings. Numbers and booleans are not supported, but in most cases a number represented as a string will be automatically converted when substituted.
+
+!include
+********
+
+- Insert the contents of another YAML file at this position.
+- Useful for splitting configurations into reusable parts.
+- Substitutions can be used in the included file to reference values passed to ``!include``. Such values will override any global substitutions, so global substitutions can be used to provide default values.
+
+Example:
+
+.. code-block:: yaml
+
+    binary_sensor:
+      - platform: gpio
+        id: button1
+        pin: GPIOXX
+        on_multi_click: !include { file: on-multi-click.yaml, vars: { id: 1 } } # inline syntax
+      - platform: gpio
+        id: button2
+        pin: GPIOXX
+        on_multi_click: !include
+          # multi-line syntax
+          file: on-multi-click.yaml
+          vars:
+            id: 2
+
+Packages
+********
+
+The ``packages`` feature allows you to define reusable and potentially partial configurations that can be included in your main configuration. Including a package file will merge its contents with your main configuration in a non-destructive way.
+
+Example:
+
+.. code-block:: yaml
+
+    # main file contents
+
+    packages:
+      wifi: !include common/wifi.yaml
+      living_room: !include common/packages/living_room.yaml
+
+    wifi:
+      ssid: "MyActualWiFi" # overrides the value in common/wifi.yaml
+
+    # The contents of common/wifi.yaml
+
+    wifi:
+      ssid: "MyWiFi"
+      password: !secret wifi_password
+
+Note that the key in a packages line is just a placeholder - it must be unique within the `packages` mapping, and should ideally be chosen to indicate its purpose, but otherwise can be anything you like that is a valid key.
+
+Variables can be provided for substitutions when including a package file, just as for regular includes.
+
+The `packages:` key can only appear at the root level of the YAML file, in other words it must start in column 1. Its location within the file is however not important. Files included under a `packages:` key will be inserted at the root level of the file, so the contents of the file should look like fragments of a standard configuration. This is in contrast to regular includes, which are inserted at the same level as the `!include` key.
```

Comment:
I suggest to be make this paragraph more general because this is also true for the `substitutions` as well.

Reply:
This is explaining the difference between using `!include` in two different contexts. `!include` is not relevant to the `substitutions:` section.